engine/graphics/font.cpp

```diff
@@ -0,0 +1,171 @@
+#include "font.h"
+
+#include "image.h"
+#include "logger.h"
+#include "utils.h"
+
+#define STB_TRUETYPE_IMPLEMENTATION
+#include <stb_truetype.h>
+
+#include <fstream>
+
+namespace pong
+{
+
+// The font will load these characters by default.
+// If a character is not in this list, it will be loaded the first time it is used.
+static const std::string DEFAULT_FONT_CHARACTERS =
+    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-=_+[]{}\\|;:'\",<.>/?`~ ";
+
+static constexpr int PIXEL_LINE_HEIGHT = 256;
+
+bool Font::LoadFont(const std::string& fontName, const std::string& fontPath)
```

Comment:
Well I guess the first issue is being able to read in a file at compile time. Im not finding a real easy way to do that, and especially not since the true type font files are not plain text either. Is there a feasable way to real in a file like that at compile time?

Reply:
Hmm, yeah that is a interesting way to do it. Im curious if that is very widely used.

Reply:
Seems a little esoteric. I think you'd have to be seriously stressed about some overhead for the user